backend/controllers/backLink.ts

```diff
@@ -0,0 +1,37 @@
+import config from '../config'
+
+const sanitizeUrl = (url: string) => (url?.endsWith('/') ? url.substring(0, url.length - 1) : url)
+
+export interface RegisteredService {
+  name: string
+  hostname: string
+  defaultBackLinkText: string
+}
+
+export const registeredBackLinkServices = [
+  {
+    name: 'digital-prison-services',
+    hostname: '',
+    defaultBackLinkText: 'View most recent search',
+  },
+  {
+    name: 'welcome-people-into-prison',
+    hostname: sanitizeUrl(config.apis.welcomePeopleIntoPrison.url),
+    defaultBackLinkText: 'Back to Welcome people into prison',
+  },
+] as Array<RegisteredService>
+
+export const saveBackLink = (registeredServices: Array<RegisteredService>) => async (req, res) => {
+  const { service, returnResource, toResource, backLinkText } = req.query
+
+  if (!service || !returnResource || !toResource) throw new Error('Required query parameters missing')
+
+  const userBackLink = registeredServices.find((e) => e.name === service)
+  if (!userBackLink) throw new Error(`Could not find service: [${service}]`)
+
+  req.session.userBackLink = {
+    url: userBackLink.hostname + returnResource,
+    text: backLinkText || userBackLink.defaultBackLinkText,
+  }
+  res.redirect(toResource)
```

Comment:
We can't allow the client to specify the location for arbitrary redirects

Reply:
Prefixed with the application url, so all redirects are scoped to be within DPS. For some reason, this is what I assume **res.redirect** did anyways 😂 Probably because I've only ever used it for that! 😁